### PR TITLE
ASoC: SOF: amd : Implementation of sof host stream pointer callback f…

### PR DESCRIPTION
**Pull request by ajitkupandey**
…unction

This patch adds pointer callback implementation for acp pcm driver.

Signed-off-by: V sujith kumar Reddy <vsujithkumar.reddy@amd.com>

### CODE REVIEW

File: sound/soc/sof/amd/acp-pcm.c [hunk 1]
```diff
@@ -83,3 +84,88 @@ int acp_pcm_close(struct snd_sof_dev *sdev, struct snd_pcm_substream *substream)
...
+	spcm = snd_sof_find_spcm_dai(scomp, rtd);
+	pcm_id = get_id_from_list(spcm->pcm.pcm_name);
+
+	bytescount = acp_get_byte_count(sdev, substream, pcm_id);
+	if (bytescount < 0)
+		return -EINVAL;
+	buffersize = frames_to_bytes(substream->runtime, substream->runtime->buffer_size);
+	pos = do_div(bytescount, buffersize);
+
+	return bytes_to_frames(substream->runtime, pos);
```
Comment (plbossart) [inline sound/soc/sof/amd/acp-pcm.c:169]: I believe this is incorrect @ajitkupandey 

the .pointer in ALSA is the host-level DMA position, ie. the DMA that is used to feed data into the DSP memory.

the link level information is very useful, but it needs to be reported with the .delay field.
The difference between the host DMA position and link DMA position is the delay.

Comment (plbossart) [reply sound/soc/sof/amd/acp-pcm.c:169]: see https://www.kernel.org/doc/html/latest/sound/designs/timestamping.html for a ascii-art that describes the different things you can report.

Comment (ajitkupandey) [reply sound/soc/sof/amd/acp-pcm.c:169]: Thanks..  we will take a look into this

**Comment by plbossart** [comment]
@ajitkupandey @reddysujith @cujomalainey  this PR has been marked as 'Unclear' for several weeks. Unless there's an update and/or a need for comments, we should close this PR. You can always resubmit a new one later, it's just not efficient to leave such PRs dangling without any progress.

**Comment by bhiregoudar** [comment]
We will close this PR and create a new PR later once we have update.

> @ajitkupandey @reddysujith @cujomalainey this PR has been marked as 'Unclear' for several weeks. Unless there's an update and/or a need for comments, we should close this PR. You can always resubmit a new one later, it's just not efficient to leave such PRs dangling without any progress.

We will close this PR and create a new PR later once we have update.

**Pull request closed without merging**